test/basic_test.exs

```diff
@@ -19,100 +19,124 @@ defmodule BasicTest do
     assert :ok = Basic.publish(meta[:chan], "", "", "ping")
   end
 
-  test "basic return", meta do
-    :ok = Basic.return(meta[:chan], self())
+  for cons <- [:selective, :direct] do
```

Comment:
It is making test's readability lower. Can we leave the existing tests as they are? Then add couple tests which test direct consumer behaviours? It can be a separate test file like `direct_consumer_test.exs`.

Reply:
Just to clarify - I am not suggesting copy paste all tests to `direct_consumer_test.exs`. Probably a good idea to write a test for direct consumer which covers open channel, send and consume a message, close a channel etc.

Reply:
Sure, will do. I just didn't want to duplicate code, as existing tests could cover both cases. I do agree though on readability argument.